lib/puppet/defaults.rb

```diff
@@ -550,6 +550,10 @@ def self.valid_file_checksum_types
         contains any characters with special meanings in URLs (as specified by RFC 3986
         section 2.2), they must be URL-encoded. (For example, `#` would become `%23`.)",
     },
+    :no_proxy => {
+      :default    => "localhost",
```

Comment:
maybe this should include `127.0.0.1`?

Reply:
no_proxy is documented as a comma-separated list of hostname suffixes, but 127.0.0.1 will match 127.0.0.1, so I'd say yes.

Reply:
Updated to include 127.0.0.1